src/controllers/krate/publish.rs

```diff
@@ -221,14 +221,34 @@ pub async fn publish(app: AppState, req: BytesRequest) -> AppResult<Json<GoodCra
             return Err(cargo_err("expected at most 5 categories per crate"));
         }
 
+        let max_features = existing_crate
+            .and_then(|c| c.max_features.map(|mf| mf as usize))
+            .unwrap_or(app.config.max_features);
+
         let features = tarball_info.manifest.features.unwrap_or_default();
+        let num_features = features.len();
+        if num_features > max_features {
+            return Err(cargo_err(&format!(
+                "Maximum number of features was reached (uploaded: {}, limit: {})",
```

Comment:
I've been wondering about that, but then I noticed that our "your crate file is too big" error message also has no such information and people are apparently clever enough to contact our support email address. I'm open to adding it though if you think it would be useful.

Reply:
I thought some more about this, and I guess I agree. I've made the error message a bit more friendly and included a sentence about sending us an email to increase the limits.